pkg/labels/selector.go

```diff
@@ -374,6 +374,10 @@ type Lexer struct {
 	pos int
 }
 
+func NewLexer(s string, pos int) *Lexer {
```

Comment:
The trouble is I will still need this for external TargetSelector (string) to internal LabelSelector now. labels.Parse() returns an interface whose underlying type is an unexported struct which in turn typedefs an exported type. However, I need LabelSelector struct. So I think this is a safe way to construct LabelSelector.